src/main/java/com/rickmorty/Services/UserService.java

```diff
@@ -0,0 +1,73 @@
+package com.rickmorty.Services;
+
+import com.rickmorty.Models.UserModel;
+import com.rickmorty.Repository.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class UserService {
+
+    @Autowired
+    private UserRepository userRepository;
+
+    // Verifica se um e-mail já está cadastrado
+    public boolean existsByEmail(String email) {
+        return userRepository.existsByEmail(email);
+    }
+
+    // Salva um novo usuário
+    public UserModel saveUser(UserModel user) {
```

Comment:
Fazer com que o método saveUser receba um userDTO, e continue a deserialização (converter de DTO para Model), e fazer a verificação; se o email já for registrado no sistema, retorna erro, se não salva no banco.

Reply:
Foi feito a devida alteração.